### PR TITLE
Enhancement 110: add tool to handle keylime policies 

### PR DESCRIPTION
**Pull request by sergio-correia**
Initial steps of enhancement #110, which is to provide a single
tool that provides the functionality of the existing tools we
have.

In this commit, we add the `keylime_policy` tool, that is currently
able to handle the creation of runtime policies.

To see the available options:

`keylime_policy --help`

To see the options available related to creating runtime policies:

`keylime_policy create runtime --help`

Some of the options available include e.g. using an IMA measurement
list, legacy allowlist/excludelist files, local/remote RPM repositories,
specifying a rootfs filesystem, initrds, a base policy, etc.

### CODE REVIEW

**Review by maugustosilva — APPROVED** [review]

**Comment by maugustosilva** [comment]
LGTM. I will confirm with @ansasaki that his PR #1573 is indeed ok (I believe it is the usual case where we can ignore `rawhide` failures), merge that one and then merge this one (after the temporary last commit is merged)

**Comment by maugustosilva** [comment]
https://github.com/keylime/keylime/pull/1573 was merged. It is now a matter of rebasing and we can merge this one. I do concur that this is a very good start for the policy creation tool.

**Comment by kkaarreell** [comment]
I think it would be nice to modify keylime-tests so that it utilize this new tool instead of old scripts. That would give us even more testing.
Also, I have noticed that codecov coverage submit has failed, and therefore code coverage details are not available in this PR.
Can anyone please re-run https://github.com/keylime/keylime/actions/runs/9763763345/job/26950409468?pr=1568 so we can see what is the actual code coverage?

**Comment by ansasaki** [comment]
> I think it would be nice to modify keylime-tests so that it utilize this new tool instead of old scripts. That would give us even more testing. Also, I have noticed that codecov coverage submit has failed, and therefore code coverage details are not available in this PR. Can anyone please re-run https://github.com/keylime/keylime/actions/runs/9763763345/job/26950409468?pr=1568 so we can see what is the actual code coverage?

I tried re-running, but if failed with the message:
```
[2024-07-15T11:47:54.503Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 429 - {'detail': ErrorDetail(string='Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 2996s.', code='throttled')}
[2024-07-15T11:47:54.504Z] ['info'] Codecov will exit with status code 0. If you are expecting a non-zero exit code, please pass in the `-Z` flag
```
Note that the test does not fail because the error code was zero even though the upload failed.

I saw the same error in https://github.com/keylime/keylime/pull/1580, so either the `CODECOV_TOKEN` is not set as a secret in the repository, or using it to avoid the rate limiting is not sufficient.

**Comment by kkaarreell** [comment]
Hm, it seems I have provided incorrect token value. Will reach out to @maugustosilva with a correct value.

**Comment by kkaarreell** [comment]
Code coverage is working again, you can rebase this PR.

**Comment by THS-on** [comment]
@ansasaki is this still blocked by something?

**Comment by ansasaki** [comment]
> @ansasaki is this still blocked by something?

AFAIK, @sergio-correia is trying to fix the pylint failures, even though those are not related with the changes in this PR.

I would suggest we ignore the pylint failures for now, move the pylint-related commits to a dedicated PR and merge this

**Comment by ansasaki** [comment]
The tests are failing because the CI image with the new test requirements was not built due to a bug in the workflow :disappointed: 

I fixed the bug in #1596, and I'm updating the base image via #1605. Hopefully after #1605 is merged, both base image and CI image are properly updated and the tests here should be fixed.

**Comment by sergio-correia** [comment]
> > @ansasaki is this still blocked by something?
> 
> AFAIK, @sergio-correia is trying to fix the pylint failures, even though those are not related with the changes in this PR.
> 
> I would suggest we ignore the pylint failures for now, move the pylint-related commits to a dedicated PR and merge this

Currently, there are 2 pyright issues, one in `ca_impl_openssl.py` and another in `web/base/controller.py`, both of which are unrelated to this PR.

**Review by ansasaki — APPROVED** [review]
The `pyright` failures are unrelated with the changes. We can fix them on a separate PR.

**Comment by ansasaki** [comment]
@THS-on I think this is ready to merge

**Comment by THS-on** [comment]
@ansasaki I'll merge it then.

@sergio-correia I think the next step is to add some kind of deprecation warning to our old scripts and update the documentation to use the new one.

@kkaarreell we then probably also need to migrate the test suite to use this script.

**Pull request merged**
